### ISSUE TITLE
SLIMIT Order by Tag

### ISSUE THREAD

**Issue by arrmo** [influxdata/influxdb#10495]
__System info:__ 
InfluxDB v1.7

__Steps to reproduce:__
I have a database full of traffic data, in and out by host. All working fine, but I want to plot / select only the top N traffic users. SLIMIT should do this, but I need to be able to order the items by traffic (summation), before SLIMIT is applied. It doesn't seem that this is possible? For example, 
`SELECT mean("value") FROM "rp2h"."rate" WHERE time > now()-2h GROUP BY time(1s), "host", "direction" LIMIT 5 SLIMIT 3`
=> I need to be able to order the data for SLIMIT, is there a way to do this?

Thanks!

**Comment by stffer** [comment]
is there a way to do this?

**Comment by arrmo** [comment]
Not that I have found ... :-(.

**Comment by bemyth** [comment]
although i don't konw how to solve your problem, but i know this is because slimit works first in sql.

**Comment by dataviruset** [comment]
I am also looking for something that could do what topk is doing in Prometheus.
https://prometheus.io/docs/prometheus/latest/querying/operators/#aggregation-operators

**Comment by arrmo** [comment]
It's unfortunate that there are no updates on this one - as a few folks seem to want similar / the same functionality. Thoughts?

Thanks!

**Comment by arrmo** [comment]
So no plans to address this?

Thanks!

**Comment by arrmo** [comment]
No, I guess close it. It would really be nice to have this, but seems it's not coming ... 😞

**Comment by Raptime** [comment]
Do we know if this is possible with the new flux query langage?
